### PR TITLE
Adds support for the configuration cache flags

### PR DESCRIPTION
**Pull request by bmuskalla**

### CODE REVIEW

File: _gradle [hunk 1]
```diff
@@ -341,6 +348,7 @@ _gradle() {
         '-Dorg.gradle.logging.level=[Set default Gradle log level.]:log level:(quiet warn lifecycle info debug)' \
         '-Dorg.gradle.parallel=[Set true to enable parallel project builds.]:(true false)' \
         '-Dorg.gradle.priority=[Set priority for Gradle worker processes.]:priority:(low normal)' \
+        '-Dorg.gradle.unsafe.configuration-cache=[Set true to enable Gradle Configuration Cache.]:enable configuration cache:(true false)' \
```
Comment (eskatos) [inline _gradle:351]: There's also `org.gradle.unsafe.configuration-cache-problems=[fail|warn]`

Comment (bmuskalla) [reply _gradle:351]: Ah sure, can add. Generally, should we even put the system properties in here given they're `unsafe` (and temporary, question mark?). Or should we only recommend the cli flags?

Comment (eskatos) [reply _gradle:351]: Properties are intended for use in `gradle.properties` so I guess we should be fine omitting them in the completion.

File: gradle-completion.bash [hunk 2]
```diff
@@ -98,6 +100,7 @@ __gradle-long-options() {
 --init-script           - Specifies an initialization script
 --max-workers           - Set the maximum number of workers that Gradle may use
 --no-build-cache        - Do not use the Gradle build cache
+--no-configuration-cache  - the configuration cache. [incubating]
```
Comment (eriwen) [inline gradle-completion.bash:103]: "Do not use the configuration cache"?

File: _gradle (hunk 1 above)
Comment (bmuskalla) [reply _gradle:351]: Makes sense, removed them completely. Out of curiosity, why are the properties called `unsafe` while the flags look like regular flags?

Comment (eskatos) [reply _gradle]: It conveys the unsafe nature of the experimental feature. And, given we want to make it enabled by default in the future, it'll make the transition easier by also using new properties without the `unsafe` part.

**Pull request merged**